File: Python/0053_maximum_subarray.py
# 2020-08-07 23:50:02

class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        max_ = -math.inf
        cur = 0

        for num in nums:
            cur = cur + num if not cur <= 0 else num
            max_ = max(cur, max_)

        return max_

File: C++/0053_maximumsubarray.cc
// 2020-08-08 23:54:28

#include <limits>
#include <vector>


class Solution {
public:
    int maxSubArray(std::vector<int>& nums) {
        int cur = 0;
        int max_ = std::numeric_limits<int>::min();

        for (auto num: nums) {
            cur = cur <= 0 ? num : cur + num;
            max_ = max(cur, max_);
        }
        return max_;
    }

    int max(int a, int b) {
        return a > b ? a : b;
    }
};